tests/Functional/GetRateTest.php

```diff
@@ -33,16 +36,12 @@ public function testGetAllRateRecords()
         $rates = FixtureFactory::createPorter()->import(new ImportSpecification(new GetAllRate($this->apiKey)))
             ->findFirstCollection();
 
-        $base = $rates->getBase();
-        $quote = $rates->getQuote();
-        $time = $rates->getTime();
-        $rate = $rates->getRate();
-
-        $this->assertSame('BTC', $base);
-        $this->assertContains('USD', $quote);
-        $this->assertContains('GBP', $quote);
-        $this->assertContains('EUR', $quote);
-        $this->assertLessThanOrEqual(time(), strtotime($time[0]));
-        $this->assertInternalType('float', $rate[0]);
+        $this->assertSame('BTC', $rates->getBase());
+
+        $rateRecords = $rates->toAssociativeArray();
+
+        $this->assertArrayHasKey('asset_id_quote', $rateRecords[0]);
+        $this->assertArrayHasKey('rate', $rateRecords[0]);
+        $this->assertArrayHasKey('time', $rateRecords[0]);
```

Comment:
This is a bad pattern for testing iterators: we should ensure the whole iterator looks and works as we expect, instead of only looking at the first record. Use `foreach` to loop over each record and assert each record is correct. It's a good idea to make assertions on the data too, not just the keys. If you don't know what the data should look like, at least try a type assertion and maybe a not empty assertion if the data should be non-null.